components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetMacros.xml

```diff
@@ -199,6 +199,45 @@
 #end## field not configured as inactive, field not empty in view mode
 #end
 
+#macro (__displayDate)
+#set ($format = $services.recordConfiguration.activeConfiguration.dateOfBirthFormat)
+#set ($definedFormat = "$!object.xWikiClass.get($name).getProperty('dateFormat').value")
+#if ($xcontext.action == 'edit')
+  #if ($value)
+    #set ($value = $services.jodatime.getMutableDateTime($value.time))
+    #set ($discard = $value.setZoneRetainFields($services.jodatime.UTCTimezone))
+  #end
+  &lt;input type='text' id='${prefix}${name}' title="$!escapetool.xml($format)" name='${prefix}${name}' class='hidden fuzzy-date' value="$!services.jodatime.getDateTimeFormatterForPattern($definedFormat).print($value)" alt="$!services.jodatime.ISODateTimeFormatter.print($value)"/&gt;&lt;input type='hidden' name='${prefix}${name}_entered' id='${prefix}${name}_entered' value="$!{escapetool.xml($object.getProperty("${name}_entered").value)}"/&gt;
+#else
+#set ($enteredValue = $object.getProperty("${name}_entered").value)
```

Comment:
I think I figured it out, as it is this macro relies on being embedded as one of `customDisplay`-s of a property, in which case $name is implicitly defined. If I'm right, it is much better to change it to be an explicit parameter and/or rename the macro to `displayPropertyAsDate()` or something like that, since it looks like all of this code relies on $name being the property name, this macro cant be used to display a date that you have in say a variable `$date`, so the name may be confusing for those who don't know the back story